### PR TITLE
cleanup: Network Throttling for Additional Networks code in DirectVifDriver.java 

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR is a small cleanup where to blocks in an `if ... else ` block do exactly the same.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/DirectVifDriver.java [hunk 1]
```diff
@@ -47,12 +47,7 @@
     public LibvirtVMDef.InterfaceDef plug(NicTO nic, String guestOsType, String nicAdapter, Map<String, String> extraConfig) throws InternalErrorException, LibvirtException {
         LibvirtVMDef.InterfaceDef intf = new LibvirtVMDef.InterfaceDef();
 
-        if (nic.getType() == Networks.TrafficType.Guest) {
-            Integer networkRateKBps = (nic.getNetworkRateMbps() != null && nic.getNetworkRateMbps().intValue() != -1) ? nic.getNetworkRateMbps().intValue() * 128 : 0;
-            intf.defDirectNet(_libvirtComputingResource.getNetworkDirectDevice(), null, nic.getMac(), getGuestNicModel(guestOsType, nicAdapter),
-                _libvirtComputingResource.getNetworkDirectSourceMode(), networkRateKBps);
-
-        } else if (nic.getType() == Networks.TrafficType.Public) {
+        if (nic.getType() == Networks.TrafficType.Guest || nic.getType() == Networks.TrafficType.Public) {
```
Comment (GutoVeronezi) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/DirectVifDriver.java:50]: We could create a static `Set`, add the types `Networks.TrafficType.Guest` and `Networks.TrafficType.Public`, and then verify if the set contains the `nic.getType()`. This way we can avoid long statements, as, if we need add another type, we just have to add it in the `Set`.

Comment (DaanHoogland) [reply plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/DirectVifDriver.java:50]: I would considder
```suggestion
        if (Sets.newHashSet(Networks.TrafficType.Guest,
                            Networks.TrafficType.Public).contains(nic.getType()) {
```
but I don't think it is an improvement with only two elements to check. Like the pattern for larger sets, but At only two elements it doesn't make it more readable, certainily not with the separate construction above the condition.

**Comment by sureshanaparti** [comment]
@blueorangutan package

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/DirectVifDriver.java [hunk 2]
```diff
@@ -47,12 +47,8 @@
     public LibvirtVMDef.InterfaceDef plug(NicTO nic, String guestOsType, String nicAdapter, Map<String, String> extraConfig) throws InternalErrorException, LibvirtException {
         LibvirtVMDef.InterfaceDef intf = new LibvirtVMDef.InterfaceDef();
 
-        if (nic.getType() == Networks.TrafficType.Guest) {
-            Integer networkRateKBps = (nic.getNetworkRateMbps() != null && nic.getNetworkRateMbps().intValue() != -1) ? nic.getNetworkRateMbps().intValue() * 128 : 0;
-            intf.defDirectNet(_libvirtComputingResource.getNetworkDirectDevice(), null, nic.getMac(), getGuestNicModel(guestOsType, nicAdapter),
-                _libvirtComputingResource.getNetworkDirectSourceMode(), networkRateKBps);
-
-        } else if (nic.getType() == Networks.TrafficType.Public) {
+        if (Sets.newHashSet(Networks.TrafficType.Guest,
```
Comment (weizhouapache) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/DirectVifDriver.java:50]: is Sets imported already ?

**Comment by DaanHoogland** [comment]
@weizhouapache @sureshanaparti the class `DirectVifBridge` is only referred from a comment in `agent.properties` and never used directly. I wonder if the real cleanup wouldn't be to remove the class entirely.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@DaanHoogland 
do we really need this fix ?

**Comment by DaanHoogland** [comment]
> @DaanHoogland do we really need this fix ?

no @weizhouapache , it is a code cleanup.

**Comment by weizhouapache** [comment]
> > @DaanHoogland do we really need this fix ?
> 
> no @weizhouapache , it is a code cleanup.

ok. got it.

**Review by weizhouapache — APPROVED** [review]

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**